From 1b95818798f0aaa3f501930159b538083bf749f1 Mon Sep 17 00:00:00 2001
From: Duncan Macleod <macleoddm@cardiff.ac.uk>
Date: Thu, 25 Jul 2024 11:43:16 +0100
Subject: [PATCH] gwpy.table: fix use of numpy types

float_ and unicode_ were aliases for a while and were removed in numpy 2

closes #1776
---
 gwpy/table/io/ligolw.py            | 2 +-
 gwpy/table/tests/test_io_ligolw.py | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gwpy/table/io/ligolw.py b/gwpy/table/io/ligolw.py
index f69976405..282e6a3bd 100644
--- a/gwpy/table/io/ligolw.py
+++ b/gwpy/table/io/ligolw.py
@@ -58,7 +58,7 @@ try:
 except ImportError:
     pass
 else:
-    NUMPY_TYPE_MAP[LIGOTimeGPS] = numpy.float_
+    NUMPY_TYPE_MAP[LIGOTimeGPS] = numpy.float64
 
 
 # -- utilities ----------------------------------------------------------------
diff --git a/gwpy/table/tests/test_io_ligolw.py b/gwpy/table/tests/test_io_ligolw.py
index f7090629a..60e17775e 100644
--- a/gwpy/table/tests/test_io_ligolw.py
+++ b/gwpy/table/tests/test_io_ligolw.py
@@ -70,7 +70,7 @@ def test_to_astropy_table_empty():
     tab = EventTable(llwtable, columns=["peak", "ifo"])
     assert set(tab.colnames) == {"peak", "ifo"}
     assert tab['peak'].dtype.type is numpy.object_
-    assert tab['ifo'].dtype.type is numpy.unicode_
+    assert tab['ifo'].dtype.type is numpy.str_
 
 
 @pytest.mark.requires("ligo.lw.lsctables")
-- 
GitLab

